### PR TITLE
#2658: Set a correct aria-describedby when sublabel and error

### LINKED ISSUE #2658: DNS Name server: ANDI output for error messages
### Current Behavior

**The `Name server` and `IP address (IPv4 or IPv6)` screen reader/ANDI output does not account for error states.**

IP address (IPv4 or IPv6)
Screenshot 2024-08-27 at 9 16 19 AM
Screenshot 2024-08-27 at 9 12 50 AM

Name server
Screenshot 2024-08-27 at 9 15 09 AM
Screenshot 2024-08-27 at 9 15 59 AM
`Name server` and `IP address (IPv4 or IPv6)` `Delete` buttons
The ANDI output does not specify which name server will be deleted.  
Screenshot 2024-08-27 at 9 35 13 AM

### Expected Behavior

The ANDI output accounts for all error states for the Name server and IP address fields.
**IP address (IPv4 or IPv6)**
- The ANDI output should be: `IP address (IPv4 or IPv6) Example: 86.124.49.54 or 2001:db8::1234:5678 | Using your domain for a name server requires an IP address`

**Name servers**
Depending on the error the ANDI output should be: 
- `Name server 2 * Example: ns2.example.com | You already entered this name server address. Name server addresses must be unique | required`
- `Name server 2 * Example: ns2.example.com | At least two name servers are required`
- `Name server 2 * Example: ns2.example.com | You must provide a name server to enter an IP address | Required`
- If there are 2 errors: `Name server 2 * Example: ns2.example.com | At least two name servers are required  | You must provide a name server to enter an IP address | required`

**`Name server` and `IP address (IPv4 or IPv6)` `Delete` buttons**
- For Name server 1 the Delete buttons ANDI output should be: `Delete name server 1`
- For Name server 2 the Delete buttons ANDI output should be: `Delete name server 2`
- And so on

### Steps to Reproduce

1. As a domain manager edit the DNS Name servers
2. Identify all error messages in the screenshots above

### Environment

Environment: development
Browser: Chrome

### Additional Context

_No response_

### Issue Links

_No response_

### Prioritization Comments
**WCAG Conformance Level:** 2.1 AA
**WCAG Standard that applies:** 
3.3.1 - Error Identification (Level A) - Critical
3.3.3 - Error Suggestion (Level AA) - Critical
4.1.2 - Name, Role, Value (Level A) - Critical
**Critical Failure if Not Met?:**  Yes
**Prioritization:** Blocker
**Reason:** Ensuring accurate screen reader output for Name Server and IP Address fields, including error states, is essential for accessibility. With the current behavior, users are unable to perceive the screen reader outputs for each error state, including unique messages for each type of error on Name Server and IP Address fields, therefore being a blocker to these users.

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #2658 

## Changes

- We actually had a bug in how we're handling all inputs with sublabel text, but it was invisible since zip inputs (the majority of such inputs) have a sulabel text and error message that match. One was not showing, but we never noticed. This PR fixes this issue, so on nameservers the aria-describedby will show sublabel text and error message. On Zip fields, you will see sublabel text and error message as well, but it will look like: Enter a 5-digit or 9-digit zip code, like `12345 or 12345-6789. Enter a 5-digit or 9-digit zip code, like 12345 or 12345-6789.` <- This would be the code working **as expected**.
- Caught a second bug that was breaking, eg, the domain manager form input and fixed it

## Context for reviewers

@cisagov/gov-designers, please read the Changes section above. This fix makes other part of the code (zip inputs) repetitive in error state on SR.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [x] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

## Screenshots

Screenshot 2024-12-09 at 6 36 10 PM

Screenshot 2024-12-09 at 6 36 17 PM

Screenshot 2024-12-09 at 6 36 24 PM

### CODE REVIEW

**Review by CocoByte — APPROVED** [review]
LGTM

File: src/registrar/templates/django/forms/widgets/input.html [hunk 1]
```diff
@@ -5,6 +5,5 @@
     class="{{ uswds_input_class }}{% if classes %} {{ classes }}{% endif %}"
     {% if widget.value != None %}value="{{ widget.value|stringformat:'s' }}"{% endif %}
     {% if aria_label %}aria-label="{{ aria_label }} {{ label }}"{% endif %}
-    {% if sublabel_text %}aria-describedby="{{ widget.attrs.id }}__sublabel"{% endif %}
```
Comment (rachidatecs) [inline src/registrar/templates/django/forms/widgets/input.html:8]: This was always overriding logic in field helpers

File: src/registrar/templatetags/field_helpers.py [hunk 2]
```diff
@@ -152,11 +156,16 @@ def input_with_errors(context, field=None):  # noqa: C901
...
+    # We handle sublabel_text here instead of directy in the template to avoid conflicts
+    if sublabel_text:
+        sublabel_div_id = f"{attrs['id']}__sublabel"
+        described_by.insert(0, sublabel_div_id)
+
     if described_by:
         # ensure we don't overwrite existing attribute value
         if "aria-describedby" in attrs:
             described_by.append(attrs["aria-describedby"])
-        attrs["aria_describedby"] = " ".join(described_by)
```
Comment (rachidatecs) [inline src/registrar/templatetags/field_helpers.py:159]: This was a seondary / unrelated bug that broke, eg, the domain manager form input

File: src/registrar/templatetags/field_helpers.py [hunk 3]
```diff
@@ -152,11 +156,16 @@ def input_with_errors(context, field=None):  # noqa: C901
     if group_classes:
         context["group_classes"] = " ".join(group_classes)
 
+    # We handle sublabel_text here instead of directy in the template to avoid conflicts
```
Comment (CocoByte) [inline src/registrar/templatetags/field_helpers.py:159]: Always appreciate comments like this

**Review by CocoByte — APPROVED** [review]

**Review by SamiyahKey — APPROVED** [review]

**Pull request merged**